rasa/core/policies/ted_policy.py

```diff
@@ -268,25 +282,47 @@ def _create_model_data(
     def _create_label_data(self, domain: Domain) -> RasaModelData:
         # encode all label_ids with policies' featurizer
         state_featurizer = self.featurizer.state_featurizer
-        all_labels = state_featurizer.create_encoded_all_actions(domain)
+        all_labels = state_featurizer.create_encoded_all_actions_sparse(domain)
         all_labels = all_labels.astype(np.float32)
 
         label_data = RasaModelData()
         label_data.add_features(LABEL_FEATURES, [all_labels])
         return label_data
 
+    def _create_label_data_e2e(self, label_data) -> RasaModelData:
+        # encode all label_ids with policies' featurizer
+        sparse_features = []
+        dense_features = []
+        for idx, feats in label_data:
+            if feats[0] is not None:
+                sparse_features.append(feats[0].astype(np.float32))
+            if feats[1] is not None:
+                dense_features.append(feats[1])
+
+        sparse_features = scipy.sparse.vstack(sparse_features)
```

Comment:
do we vstack sparse features to get rid of middle dimension of `1`?

Reply:
No, that's because we want to store all label features into one array, so that the rest of the logic in model_data and further down the line.